### PR TITLE
Add cookbook entry for Database ORM / Eloquent usage

### PR DESCRIPTION
**Pull request by svpernova09**
Added cookbook page detailing how I implemented Eloquent.

Feedback/suggestions welcome.

### CODE REVIEW

File: docs/cookbook/database-orm.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    protected $widgets;
+
+    public function __construct(Twig $view, LoggerInterface $logger, $widgets)
+    {
+        $this->view = $view;
+        $this->logger = $logger;
+        $this->widgets = $widgets;
+    }
+
+    public function __invoke(Request $request, Response $response, $args)
```
Comment (geggleto) [inline docs/cookbook/database-orm.md:104]: Please rename `$widgets` to maybe `$table` ... I would also add a type-hint to it as well. I would change the class property to match. $widgets does not accurately describe what it is.

File: docs/cookbook/database-orm.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+{% endhighlight %}
+<figcaption>Figure 4: Sample controller querying the table.</figcaption>
+</figure>
+
+### Query the table with where
+
+<figure>
+{% highlight php %}
+...
+$widgets = $this->widgets->where('name', 'like', '%foo%')->get();
```
Comment (geggleto) [inline docs/cookbook/database-orm.md]: `$records` would be a better name for the output of the where call.

File: docs/cookbook/database-orm.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+{% endhighlight %}
+<figcaption>Figure 5: Query searching for names matching foo.</figcaption>
+</figure>
+
+### Query the table by id
+
+<figure>
+{% highlight php %}
+...
+$widgets = $this->widgets->find(1);
```
Comment (geggleto) [inline docs/cookbook/database-orm.md]: `$record` instead of `$widgets`

File: docs/cookbook/database-orm.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+{% endhighlight %}
+<figcaption>Figure 2: Configure Eloquent.</figcaption>
+</figure>
+
+## Pass a controller an instance of your table
+
+<figure>
+{% highlight php %}
+$container[App\WidgetController::class] = function ($c) {
+    $model = $c->get('db')->table('table_name');
```
Comment (geggleto) [inline docs/cookbook/database-orm.md]: I would rename this to `$table` since it is not an Eloquent Model

File: docs/cookbook/database-orm.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    return new \App\WidgetController($c->get('view'), $c->get('logger'), $model);
```
Comment (geggleto) [inline docs/cookbook/database-orm.md]: And then update it here, `$table`

File: docs/cookbook/database-orm.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+<figure>
+{% highlight php %}
+// -----------------------------------------------------------------------------
+// Service factories
+// -----------------------------------------------------------------------------
+
+$container['db'] = function ($container) {
+    $capsule = new \Illuminate\Database\Capsule\Manager;
+    $capsule->addConnection([
```
Comment (geggleto) [inline docs/cookbook/database-orm.md]: might be useful to show how you can use Slim's settings array to contain this.

``` php
$container['settings']['db'] = [
        'driver' => 'mysql',
         'host' => 'localhost'
         'database' => 'database',
         'username' => 'user',
         'password' => 'password',
         'charset'   => 'utf8',
         'collation' => 'utf8_unicode_ci',
         'prefix'    => '',
];
```

Then it's simple to do, `$capsule->addConnection($container['settings']['db']);`

**Comment by geggleto** [comment]
Sorry for nit-picking :)

**Comment by svpernova09** [comment]
No worries, thanks for the feedback. Definitely happier with the suggested changes/

**Comment by geggleto** [comment]
:+1: Looks a lot better to me :)

**Comment by akrabat** [comment]
Thanks Joe!

**Pull request merged**